code/game/objects/structures/xeno.dm

```diff
@@ -709,6 +718,13 @@ TUNNEL
 	else
 		to_chat(M, "<span class='warning'>Our crawling was interrupted!</span>")
 
+/obj/structure/tunnel/proc/hud_set_xeno_tunnel() //called by the Dig Tunnel ability after creation; make this handy overlay exclusively visible to xenos of the same hive.
+	var/image/tunneloverlay = image('icons/mob/hud.dmi', src, "hudtraitor")
+
+	for(var/mob/living/carbon/xenomorph/X in creator.hive.get_all_xenos()) //Set hive specific image visibility
+		X.client.images += tunneloverlay
+
```

Comment:
use the hud system you updated pls
If you need more pointers look at diagnostic huds in the tg codebase

Reply:
Per our discussion on discord structures seem particularly resistant to HUD implementation. I'm hoping we can find a way in the future however.